### DISCUSSION TITLE
Testing assumptions for running adonis2 over continuous data

### DISCUSSION THREAD

**Discussion by NThie** [vegandevs/vegan#516] [Q&A] [answered]
Hi,
As I understand from the documentation, adonis2 function can be used to test community differences between groups as well as over continuous data. When testing between groups, there is the assumption that the variance / dispersal needs to be equal between the two groups to properly use adonis2 - which can be checked with the betadisper function. However, betadisper does not work with continuous data - so how do I check the assumptions for using adonis2 with continuous data?

Thanks for your help!

**Comment by gavinsimpson** [comment]
The idea behind `betadisper()` is as a multivariate analogue for univariate tests of homogeneity of *group* variance such as Bartlet's or Levene's tests. I suppose this is because the model `adonis2()` fits was originally described as a multivariate analogue of ANOVA, and ANOVA is for categorical covariates.

Homogeneity of variances is harder to check within a linear model framework; one often resorts to graphical techniques like scatter plots of residuals or QQ plots.

I suppose you could artificially create groups by cutting the continuous covariate in `k` group using `cut()` and that way check if each bin had roughly the same variance by passing the created factor to `betadisper()`.

**Comment by jarioksa** [comment, marked as answer]
There is no direct way of finding that information directly from `adonis2`. However, you can build a corresponging `dbrda` model and use tools to analyse its residuals:
```r
library(vegan)
data(mite, mite.env)
mad2 <- adonis2(dist(decostand(mite, "hell")) ~ SubsDens + WatrCont, mite.env)
mdb <- dbrda(dist(decostand(mite, "hell")) ~ SubsDens + WatrCont, mite.env)
ordiresids(mdb, residuals="working")
ordiresids(mdb, "scale", residuals="working")
ordiresids(mdb, "qq", residuals="working")
```
The internal mathematical models are identical in `dbrda` and `adonis2` justifying this approach. However, the residuals concern internal ("working") form of dissimilarities (Gower double-centred dissimilarities), and for _N_ observations there are _N_(_N_–1)/2 distinct points in the plot. The plots are similar as the diagnostic plots for linear model (`lm`).

**Comment by jarioksa** [comment]
Here a simple test to see how adequate the `ordiresids` approach may be.  Let's  see if residuals of `dbrda` are similar to `betadisper` distances to group centroids.

```r
library(vegan)
data(mite, mite.env)
mbb <- betadisper(vegdist(sqrt(mite)), mite.env$Shrub, type="centroid") # default type = "median" is worse
mdb <- dbrda(vegdist(sqrt(mite)) ~ Shrub, mite.env)
dbb <- mbb$distances # distances to betadisper centroids
dw <- residuals(mdb, type="working")
dr <- as.matrix(residuals(mdb, type="response")) # needs as.matrix
plot(dbb, sqrt(rowSums(dw^2)))  # not quite similar
plot(dbb, sqrt(rowSums(dr^2))) # not quite linear, but exact mapping
```
So this would indicate that response residuals are more or less similar to `betadisper` distances whereas the internal forms are not. However, my gut feeling is that working residuals (on which the methods actually work!) are the ones that should be OK (and this conjecture rather has consequences to `betadisper`). You can also specify `type = "response"` in `ordiresids`, but for `dbrda` object you get a harmless warning as the response residuals are not returned as a symmetric square matrix, but as distances.
